### DISCUSSION TITLE
OKD Single Node Cluster

### DISCUSSION THREAD

**Discussion by crbednarz** [okd-project/okd#1216] [Q&A] [answered]
Hello,

I'm trying to determine whether or not OKD supports installing a single node cluster as a single VM, and seem to have gotten a bit lost along the way.

From looking at the official documentation [here](https://docs.okd.io/latest/installing/installing_sno/install-sno-preparing-to-install-sno.html) and [here](https://docs.okd.io/latest/installing/installing_sno/install-sno-installing-sno.html), it seems an ISO can be manually created which you could install onto a machine and have it act as the complete cluster. However, these instructions have led to a number of questions:

1. Are single-node clusters supported with OKD? Discussion in #1012 seems to indicate that OKD doesn't support single-node clusters at all without work arounds, however the [FAQ](https://github.com/openshift/okd/blob/master/FAQ.md#can-i-run-a-single-node-cluster) and docs links above seem to indicate it is.
2. The documentation for single-node OKD appears identical to single-node OpenShift, with all of the URLs directly referencing OpenShift tools. Does this work for OKD, or do they need to be replaced with those found on the OKD releases page?
3. Assuming single-node clusters are supported, the majority of documentation and discussion around them seems to be for bare-metal installations. Are there any restrictions with regards to running inside of a VM?

**Comment by vrutkovs** [comment, marked as answer]
>Are single-node clusters supported with OKD?

Yes

>The documentation for single-node OKD appears identical to single-node OpenShift

This is a bug, we need to update the docs to use https://github.com/openshift/assisted-service/tree/master/deploy/podman#okd-configuration instead of hosted Assisted Installer version

> Are there any restrictions with regards to running inside of a VM?

No, not that we know of at least

**Reply by crbednarz** [reply]
Thank you so much. The assisted installer setup you linked works quite well, and was certainly easier than trying to manually build the iso.

**Comment by mburke5678** [comment]
@crbednarz @vrutkovs Please let me know what needs to be removed and what to replace it with and I will get a docs PR going to make the changes.
